sound/soc/sof/intel/hda.c

```diff
@@ -79,10 +78,26 @@ static u32 hda_get_interface_mask(struct snd_sof_dev *sdev)
 	default:
 		break;
 	}
+}
+
+static u32 hda_get_interface_mask(struct snd_sof_dev *sdev)
+{
+	u32 interface_mask[2] = { 0 };
+
+	hda_get_interfaces(sdev, interface_mask);
 
 	return interface_mask[sdev->dspless_mode_selected];
 }
 
+bool hda_is_dai_supported(struct snd_sof_dev *sdev, u32 dai_type, bool dspless_mode)
```

Comment:
I see, ChainDMA can only be used on links which could be used in DSPless mode, both the host and link DMA is HD-DMA.

But is this a universal rule?

I'm not sure how this can be detached DSPless mode, perhaps `is_dai_feature_supported(sdev, dai_type, u8 feature)`
enum sof_dai_feature {
	sof_dai_feature_chain_dma,
};

if the dai_type is not supported, return false.
if the feature is not  supported by the dai type, return false.

I believe that there will be comment about the connection between ChainDMA and DSPless mode, but I'm not sure about the tying of the two.

Or just have a callback as blunt as: `is_chain_dma_supported(sdev, dai_type);`

Reply:
done, see update.